httpserver/rust/src/lib.rs

```diff
@@ -26,10 +26,48 @@ impl HttpResponse {
         let body = serde_json::to_string(&payload)
             .map_err(|e| RpcError::Ser(e.to_string()))?
             .into_bytes();
+        let mut header = HeaderMap::new();
+        header.insert(
+            "Content-Type".to_string(),
+            vec!["application/json".to_string()],
+        );
         Ok(HttpResponse {
             body,
             status_code,
-            ..Default::default()
+            header,
+        })
+    }
+
+    /// Creates a response with a given status code, JSON-serialized payload, and headers specified by the header argument. Automatically includes the appropriate Content-Type header
+    ///
+    /// # Arguments
+    ///
+    /// * `payload` - Any struct implementing the Serialize trait from serde/serde_json
+    /// * `status_code` - A 16-bit unsigned integer representing the outbound HTTP status code, e.g. 200 for successful interactions, 404 for not found, etc..
+    /// * `headers` - A std::collections::HashMap from a String to a Vec() of Strings. The Content-Type header is ignored, for Content-Type will always default to application/json
+    ///
+    pub fn json_with_headers<T>(
+        payload: T,
+        status_code: u16,
+        headers: std::collections::HashMap<String, Vec<String>>,
+    ) -> Result<HttpResponse, RpcError>
+    where
+        T: Serialize,
+    {
+        let body = serde_json::to_string(&payload)
+            .map_err(|e| RpcError::Ser(e.to_string()))?
+            .into_bytes();
+
+        let mut fixed_header = headers.clone();
+        fixed_header.retain(|k, _| k.to_lowercase() != "content-type");
```

Comment:
I suggest changing the logic of this so that it only sets content-type if it's not already defined. In the case you're trying to optimize for, it's not set by the caller, and this function sets it. There could be other cases where somebody wants a json body but they also want to customize the content-type header. For example, they might want to set it to "application/json; charset=UTF-8", or the name of an application protocol based on json, etc.

Reply:
Good point. I will go ahead and make the change now.

Reply:
Added the change. I've also made the headers lowercase according to HTTP/2 standards, though input correction is performed at the moment instead of input rejection if the developer supplies capitalized headers by mistake.